src/auro-menu.js

```diff
@@ -35,12 +35,70 @@ class AuroMenu extends LitElement {
     `;
   }
 
-  // function that renders the HTML and CSS into  the scope of the component
+  firstUpdated() {
+    const options = this.shadowRoot.querySelector('slot[name=listOfOptions]').assignedNodes();
+
+    const dispatchEventOptionSelected = (el) => {
+      el.dispatchEvent(new CustomEvent('optionSelected', {
+        bubbles: true,
+        cancelable: false,
+        composed: true,
+        detail: {
+          index: el.getAttribute('index'),
+          value: el.getAttribute('value'),
+          displayText: el.innerText,
+        }
+      }));
+
+      options.forEach((option) => {
+        if (el.getAttribute('index') === option.getAttribute('index')) {
+          option.setAttribute('selected', true);
+        } else {
+          option.removeAttribute('selected');
+        }
+      });
+
+      const elIndex = parseInt(el.getAttribute('index'), 10);
+
+      for (let i = 0; i < options.length; i += 1) {
+        if (elIndex === i) {
+          options[i].setAttribute('selected', '');
+          options[i].querySelector('span').setAttribute('style', 'visibility: visible; margin-right: 8px;');
```

Comment:
refactor for later - setting a class that contains these attributes would be preferred

Reply:
Added a TODO comment above saying to refactor.